src/Symfony/Component/Console/Application.php

```diff
@@ -880,6 +882,16 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             return $command->run($input, $output);
         }
 
+        // bind before the console.command event, so the listeners have access to input options/arguments
+        try {
+            $command->mergeApplicationDefinition();
+            $input->bind($command->getDefinition());
+        } catch (InvalidArgumentException $e) {
```

Comment:
similarly https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Console/Command/Command.php#L234 should be updated